src/ai/backend/common/validators.py

```diff
@@ -465,12 +465,14 @@ def check_and_return(self, value: Any) -> datetime.timedelta:
 
 class Slug(t.Trafaret, metaclass=StringLengthMeta):
 
-    _rx_slug = re.compile(r'^[a-zA-Z0-9]([a-zA-Z0-9._-]*[a-zA-Z0-9])?$')
+    _rx_slug_ascii_only = re.compile(r'^[\w\-_.\s]+$', re.ASCII)
+    _rx_slug = re.compile(r'^[\w\-_.\s]+$')
```

Comment:
It additionally allows whitespaces compared to the previous version.
Please make it to stand out in the news fragment, and reflect that change in the test cases as well.
Also the code has "ascii_only" argument but the fragment says "unicode" mode -- please make them consistent.

Reply:
Let's update as follows:

Update the conditions for `Slug`:
 - disallow the characters outside the regex `[\w\-.]` (_no_ whitespace allowed!)
 - disallow consecutive dots such as `..`, `...`, ... in any position
 - disallow zero-length (empty) strings

A new validator `DirectoryNameComponent`:
 - disallow the characters outside the regex `[\w\s\-.]` (whitespace allowed!)
 - disallow consecutive dots such as `..`, `...`, ... in any position
 - disallow zero-length (empty) strings